File: rental-backend/rental/src/main/java/com/example/rental/entety/Reservation.java
package com.example.rental.entety;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "reservation")
@Data
@NoArgsConstructor
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "reserved_at")
    private LocalDateTime reservedAt;
    @Column(name = "expires_at")
    private LocalDateTime expiresAt;

    @ManyToOne
    @JoinColumn(name = "property_id")
    private Property property;

    @ManyToOne
    @JoinColumn(name = "client_id")
    private Client client;
}

File: rental frontend/rental/src/app/model/reservation.ts
import { Client } from "./client";
import { Property } from "./property";

export interface Reservation {
    id?: number;
    propertyId: number;
    clientId: number;
    reservedAt: Date;
    expiresAt: Date;
}

export interface ReservationRes {
    id: number;
    property: Property;
    client: Client;
    reservedAt: Date;
    expiresAt: Date;
}